### ISSUE TITLE
fetching backup pg_basebackup exited ERROR

### ISSUE THREAD

**Issue by ahmetmelihbasbug** [zalando/patroni#2200]
Hi.

When we add second machine to patroni cluster, it copy tablespace disk first and patroni data directory last.
However after finishing tablespace moving, it stop to copy main data directory (datadir).
After i try again, it could be copy successfuly (maybe)... sometimes i try 3 or 4 times.

Why patroni stop syncronization and not continue?
After copy tablespace from primary, how can i prevent pg_basebackup from "Error when fetching backup: pg_basebackup exited wit code .... failed to bootstrap from leader..... removing data /data/patroni/" ?
Thanks

**Comment by CyberDem0n** [comment]
Patroni only calls pg_basebackup or custom [replica creation method](https://patroni.readthedocs.io/en/latest/replica_bootstrap.html#building-replicas).
If it fails (exit-code != 0), what else we can do rather than removing PGDATA and retrying?

**Comment by ahmetmelihbasbug** [comment]
And I can not see any more details about failing so i can not see reason.

**Comment by CyberDem0n** [comment]
Patroni doesn't intercept stdout/stderr of pg_basebackup.
I have no clue how you configured logging. It could be that whatever pg_basebackup writes ends up I a file that is different from Patroni logs.

**Comment by ahmetmelihbasbug** [comment]
There are postgresql-%a.log with log collector option and patroni.log on the system. I can not see any result on logs of postgre and patroni .

**Comment by CyberDem0n** [comment]
If don't know how to debug something - start Patroni manually from the terminal

**Comment by ahmetmelihbasbug** [comment]
I added pg_basebackup verbose parameters on patroni config.yml then I reinitialized replica, I could see % of migration from leader to replica.
there were no errors occured. However when i list;
 
patronictl -c /etc/patroni/config.yml list -W -e

```
cLuster: postgres
Member   Host        Role            TL         Lag in MB      Tags 
--------- | --------- | --------- | --------- | --------- | --------------|
postgre1|               |  Replica  |   105      |     0        |  osync: false |
postgre2|               |  Leader   |   105      |               |                     |
```

I do not understand why "osync: false" had been  occured.  

In patroni log, I see only INFO Lock owner I am leader.. etc etc..
And secondary tells it is replica following leader. about to wait on key from leader index *******

My collagues told that there is no different things in postgresql logs... First time i see osync: false tag...
